js/index.js

```diff
@@ -1,42 +1,49 @@
 // ITERATION 1
-
 function updateSubtotal(product) {
-  console.log('Calculating subtotal, yey!');
-
-  //... your code goes here
+  let price = product.querySelector(".price span");
+  let quantity = product.querySelector(".quantity");
+  let priceVal = price.innerHTML;
+  let quantityVal = product.querySelector("input").value;
+  let subtotalVal = priceVal * quantityVal;
+  let subtotal = product.querySelector(".subtotal span");
+  subtotal.innerHTML = subtotalVal;
+  return subtotalVal;
 }
-
+// ITERATION 2
 function calculateAll() {
-  // code in the following two lines is added just for testing purposes.
-  // it runs when only iteration 1 is completed. at later point, it can be removed.
-  const singleProduct = document.querySelector('.product');
-  updateSubtotal(singleProduct);
-  // end of test
-
-  // ITERATION 2
-  //... your code goes here
-
+  let totalPrice = 0;
+  let products = document.querySelectorAll(".product");
+  products.forEach((prod) => {
+    totalPrice += updateSubtotal(prod);
+  });
   // ITERATION 3
-  //... your code goes here
+  document.querySelector("#total-value span").innerText = totalPrice;
 }
-
 // ITERATION 4
-
 function removeProduct(event) {
   const target = event.currentTarget;
-  console.log('The target in remove is:', target);
-  //... your code goes here
+  console.log("The target in remove is:", target);
+  target.parentNode.parentNode.remove();
+  calculateAll();
 }
 
 // ITERATION 5
 
-function createProduct() {
-  //... your code goes here
+function createProduct(productName, priceVal) {
+  let newProd = document.createElement("tr");
+  let productName = newProd.classList.add("product");
```

Comment:
I believe this variable declaration causes an error, as productName is already declared as an argument.